pkg/cloudevents/client/client.go

```diff
@@ -47,27 +50,40 @@ func (c *ceClient) Send(ctx context.Context, event cloudevents.Event) error {
 	}
 	// Validate the event conforms to the CloudEvents Spec.
 	if err := event.Validate(); err != nil {
-		return err
+		return nil, err
 	}
 	// Send the event over the transport.
 	return c.transport.Send(ctx, event)
 }
 
-func (c *ceClient) Receive(event cloudevents.Event) {
-	if c.receiver != nil {
-		c.receiver(event)
+// Receive is called from from the transport on event delivery.
+func (c *ceClient) Receive(ctx context.Context, event cloudevents.Event, resp *cloudevents.EventResponse) error {
+	if c.receive != nil {
+		err := c.receive(ctx, event, resp)
```

Comment:
client.receive should be mappable to an event type or other interesting filtering. Look into this.